### PR TITLE
RPC to fetch supported PHY versions for Band ID

### LINKED ISSUE #4518: Expose `band` pkg via API to enable sourcing FP constraints
#### Summary
We need to come up with a way to expose the restrictions of frequency plans with regards to MAC/PHY versions.

#### Why do we need this?
E.g. the Australian FPs do not support MAC/PHY 1.0.0. However, the Console still allows setting this combination because the frontend is unaware of such restrictions. This leads to frustration when users fill out the end device registration form and hitting this limitation only upon submitting.

image

#### What is already there? What do you see now?
- End device registration form

#### What is missing? What do you want to see?
- API to query from the Console to get the `band -> regional parameters version -> available` mapping

#### How do you propose to implement this?
Probably trivial. cc @adriansmares 

#### How do you propose to test this?
Manual testing and updating e2es.

#### Can you do this yourself and submit a Pull Request?
Need help from @adriansmares (or whoever else feels confident with this).

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

RPC to fetch supported PHY versions for Band ID. Closes #4518 

#### Changes

- Add a new `GetSupportedPhyVersions` to the configuration server.
- Implement RPC and add tests.

#### Testing

UT

##### Regressions

NA

#### Notes for Reviewers

The implementation can change a little after https://github.com/TheThingsNetwork/lorawan-stack/pull/4591 but the API will not.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by adriansmares — COMMENTED** [review]
This LGTM overall, just one question:

Do we want to manually fetch the versions of the band individually, or should we have a RPC that returns _all_ bands and their associated versions ?

cc @kschiffer

**Comment by kschiffer** [comment]
> Do we want to manually fetch the versions of the band individually, or should we have a RPC that returns _all_ bands and their associated versions ?

Yeah, so in the Console the user selects the MAC and PHY version before selecting the FP. So to be able to use this RPC, we'd have to change the order of the fields. I think this is a good idea anyway given the structure of the constraints.

Being able to retrieve all bands and associated versions at once would save us a request whenever the selection changes, making the form a bit more responsive. So if it's easy to add, I'd prefer having that as well.

**Comment by KrishnaIyer** [comment]
> Being able to retrieve all bands and associated versions at once would save us a request whenever the selection changes, making the form a bit more responsive. So if it's easy to add, I'd prefer having that as well.

@kschiffer If I understand you correctly, you'd want an additional RPC to fetch all bands or modify this one to do that?

Either one is quite easy to do so I'll update the PR based on what works best for you.

**Comment by kschiffer** [comment]
With regards to the usage for the Console, just having an RPC to retrieve all bands and associated versions is sufficient.

**Comment by KrishnaIyer** [comment]
Ok thanks. That's done now

**Review by adriansmares — APPROVED** [review]

File: pkg/band/rpc.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+package band
+
+import (
+	"context"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+// TODO: Replace this with the `band.All` logic in https://github.com/TheThingsNetwork/lorawan-stack/pull/4591
```
Comment (htdvisser) [inline pkg/band/rpc.go:23]: Please add a note on that PR to actually do this.

**Review by htdvisser — COMMENTED** [review]
LGTM

**Review by htdvisser — APPROVED** [review]
LGTM

**Pull request merged**